io.c

```diff
@@ -60,6 +60,12 @@ int uev_io_init(uev_ctx_t *ctx, uev_t *w, uev_cb_t *cb, void *arg, int fd, int e
  */
 int uev_io_set(uev_t *w, int fd, int events)
 {
+
+  if( events & UEV_ONESHOT 
+      && 1 == w->active ) {
+    return uev_watcher_rearm(w);
+  }
+
```

Comment:
Looks better, although that is not Linux KNF syntax for the `if ()` statment block.  Should look like this:

    if ((events & UEV_ONESHOT) w->active) { ...

Notice the whitespace differences.  It may seem like nitpicking, but keeping the new code additions in line with the rest is very important to most maintainers :-)